lib/src/rules/avoid_init_to_null.dart

```diff
@@ -88,8 +90,13 @@ class _Visitor extends SimpleAstVisitor<void> {
   @override
   void visitDefaultFormalParameter(DefaultFormalParameter node) {
     var declaredElement = node.declaredElement;
-    if (declaredElement == null) {
-      return;
+    if (declaredElement == null) return;
+    if (declaredElement is DefaultSuperFormalParameterElementImpl) {
```

Comment:
I guess there is no need to access the implementation, we can use `SuperFormalParameterElement` and `FieldFormalParameterElement`

Reply:
I can't easily see which APIs you need from `DefaultSuperFormalParameterElementImpl` and `DefaultFieldFormalParameterElementImpl` that aren't public APIs, but you might not need to reference them anymore.

You might consider combining the `if`s on lines 93 and 94 as
```
if (declaredElement is! DefaultSuperFormalParameterElementImpl) return;
```
just to make the code a little shorter.

Reply:
Thanks!